velox/functions/sparksql/tests/SparkPartitionIdTest.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SparkPartitionIdTest : public SparkFunctionBaseTest {
+ protected:
+  void testSparkPartitionId(
+      const std::string& expression,
+      const int partitionId) {
+    queryCtx_->testingOverrideConfigUnsafe(
+        {{core::QueryConfig::kSparkPartitionId, std::to_string(partitionId)}});
+    auto result = evaluate<ConstantVector<int32_t>>(
+        expression, makeRowVector(ROW({}), 1));
+    velox::test::assertEqualVectors(makeConstant(partitionId, 1), result);
+  }
+
+  void testSparkPartitionIdFails(
+      const std::string& expression,
+      const std::string& errorMsg) {
+    VELOX_ASSERT_THROW(
+        evaluate<ConstantVector<int32_t>>(
+            expression, makeRowVector(ROW({}), 1)),
+        errorMsg);
+  }
+};
+
+TEST_F(SparkPartitionIdTest, basics) {
+  testSparkPartitionId("spark_partition_id()", 0);
```

Comment:
The first argument to testSparkPartitionId is always the same. Hence, it is not needed and the value can be hard-coded inside testSparkPartitionId.

Reply:
Updated, thanks.